### PR TITLE
Fixes: Notifications Pending Intent Mutability issue

### LINKED ISSUE #17465: RuntimeException: Unable to unbind to service org.wordpress.android.login.LoginWpcomService@2489de8 with Intent { c...
**Sentry Url**: https://sentry.io/share/issue/5d6b5c89bb124cf08e3c13e9f35bf761/
**User Count**: 87
**Count**: 1769
**First Release**: 21.0-rc-2
**First Seen**: 2022-10-18T02:10:48.813000Z
**Last Seen**: 2022-11-15T10:51:55Z
**24 Hours**: 139
**30 Days**: 1758

```
RuntimeException: Unable to unbind to service org.wordpress.android.login.LoginWpcomService@2489de8 with Intent { cmp=org.wordpress.android/.login.LoginWpcomService }: java.lang.IllegalArgumentException: org.wordpress.android: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
     at org.wordpress.android.util.AutoForegroundNotification.getNotificationBuilder(AutoForegroundNotification.java:45)
     at org.wordpress.android.util.AutoForegroundNotification.progress(AutoForegroundNotification.java:55)
     at org.wordpress.android.login.LoginWpcomService$LoginNotification.progress(LoginWpcomService.java:122)
     at org.wordpress.android.login.LoginWpcomService.getNotification(LoginWpcomService.java:206)
     at org.wordpress.android.login.LoginWpcomService.getNotification(LoginWpcomService.java:42)
     at org.wordpress.android.util.AutoForeground.promoteForeground(AutoForeground.java:153)
     at org.wordpress.android.util.AutoForeground.onUnbind(AutoForeground.java:131)
...
(11 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by AjeshRPai**
Fixes #17465 

Dependant fluxC PR - https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/112

Although the sentry crash was pointing to the `LoginWpcomService`, the root cause for the crash was the pending intent created in `AutoForegroundNotification.java`. The Intent created in AutoForegroundNotification.java is used in the site creation service as well. 

I couldn't reproduce the bug on the login flow. But I was able to reproduce it on the site creation flow with the steps below. 

Steps to reproduce the crash 
- Launch app on Android 12 device 
- Create any WordPress site (Site picker -> + -> Create WordPress site -> Select theme -> Choose domain 
- When the message Site creation progress screen is shown put the app in the background
- Notice the log message in terminal and the notifications for site created is not shown

<details>
<summary><h3>Log message</h3></summary>

``
java.lang.RuntimeException: Unable to unbind to service org.wordpress.android.ui.sitecreation.services.SiteCreationService@d918779 with Intent { cmp=org.wordpress.android.prealpha/org.wordpress.android.ui.sitecreation.services.SiteCreationService }: java.lang.IllegalArgumentException: org.wordpress.android.prealpha: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
                                                                            Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
                                                                            	at android.app.ActivityThread.handleUnbindService(ActivityThread.java:4590)
                                                                            	at android.app.ActivityThread.-$$Nest$mhandleUnbindService(Unknown Source:0)
                                                                            	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2172)
                                                                            	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                            	at android.os.Looper.loopOnce(Looper.java:201)
                                                                            	at android.os.Looper.loop(Looper.java:288)
                                                                            	at android.app.ActivityThread.main(ActivityThread.java:7898)
                                                                            	at java.lang.reflect.Method.invoke(Native Method)
                                                                            	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
                                                                            	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
                                                                            Caused by: java.lang.IllegalArgumentException: org.wordpress.android.prealpha: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
                                                                            Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
                                                                            	at android.app.PendingIntent.checkFlags(PendingIntent.java:401)
                                                                            	at android.app.PendingIntent.getActivityAsUser(PendingIntent.java:484)
                                                                            	at android.app.PendingIntent.getActivity(PendingIntent.java:470)
                                                                            	at android.app.PendingIntent.getActivity(PendingIntent.java:434)
                                                                            	at org.wordpress.android.util.AutoForegroundNotification.getNotificationBuilder(AutoForegroundNotification.java:45)
                                                                            	at org.wordpress.android.util.AutoForegroundNotification.progressIndeterminate(AutoForegroundNotification.java:63)
                                                                            	at org.wordpress.android.ui.sitecreation.services.SiteCreationServiceNotification.createCreatingSiteNotification(SiteCreationServiceNotification.kt:14)
                                                                            	at org.wordpress.android.ui.sitecreation.services.SiteCreationService.getNotification(SiteCreationService.kt:71)
                                                                            	at org.wordpress.android.ui.sitecreation.services.SiteCreationService.getNotification(SiteCreationService.kt:25)
                                                                            	at org.wordpress.android.util.AutoForeground.promoteForeground(AutoForeground.java:153)
                                                                            	at org.wordpress.android.util.AutoForeground.onUnbind(AutoForeground.java:131)
                                                                            	at android.app.ActivityThread.handleUnbindService(ActivityThread.java:4574)
                                                                            	... 9 more
``
</details>

After fix
- Video capture showing the site created notification 

https://user-images.githubusercontent.com/17463767/202568342-87d68b99-6957-45d4-a32b-d7fc50534edd.mp4

To test:
- Test whether the pending mutability error message is shown on logcat in error mode 
- Test whether the site-created notification is shown properly 

Merge Instructions 
- Merge the Dependant fluxC PR 
- Update Utils library version 
- Remove Not Ready for Merge label 
- Merge as usual 

## Regression Notes
1. Potential unintended areas of impact
Notifications not shown properly 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by AjeshRPai** [comment]
@ovitrif 

Do you think we should target any release branches or is it fine if we target the next code freeze? The [dependant fluxC PR](https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/112) of this PR has a dependency on the  WordPress-Utils-Android to update minSdk version to 24, since the flag requires API level 23. So I am not sure whether we can push this change as a hotfix.

**Comment by ovitrif** [comment]
> Do you think we should target any release branches or is it fine if we target the next code freeze? The [dependant fluxC PR](https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/112) of this PR has a dependency on the WordPress-Utils-Android to update minSdk version to 24, since the flag requires API level 23. So I am not sure whether we can push this change as a hotfix.

Hey @AjeshRPai 👋 !
Thank you for further investigating this issue and working on this great fix 🏅 

Given the better understanding brought by your investigation into when this issue reproduces, my understanding is that it occurs only when one puts the app in background, either during login (probably with a slower internet connection?) or during site creation — which is much easier to repro — it looks rather like an edge case to me, therefore **we can target the next code freeze**.

**Review by ovitrif — APPROVED** [review]
Awesome work further investigating the issue and discovering a clear repro scenario, as well as providing the fix @AjeshRPai 🙇 🏅 🚀 

I was able to reproduce it consistently with your findings on `trunk` using Android 12+:

- ✅ Test whether the pending mutability error message is shown on logcat in error mode

  <details>
    <summary>Error Summary Log</summary>

    ```
    E/AndroidRuntime: FATAL EXCEPTION: main
        Process: com.jetpack.android.beta, PID: 5733
        java.lang.RuntimeException: Unable to unbind to service org.wordpress.android.ui.sitecreation.services.SiteCreationService@5688671 with Intent { cmp=com.jetpack.android.beta/org.wordpress.android.ui.sitecreation.services.SiteCreationService }: java.lang.IllegalArgumentException: com.jetpack.android.beta: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
        Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
  
   ```
   </details>

After checking out the branch of this PR, I could see it succeed as described:
- ✅ Test whether the site-created notification is shown properly
  <details>
    <summary>Site Created Notification</summary>

    

  </details>

**Comment by AjeshRPai** [comment]
@ovitrif 👋🏼 

Thanks for reviewing the PR. Could you approve the Dependant Utils PR? Once the Utils PR is merged I will update the Utils version of this PR and merge it as usual.

**Comment by ovitrif** [comment]
> Thanks for reviewing the PR. Could you approve the Dependant Utils PR? Once the Utils PR is merged I will update the Utils version of this PR and merge it as usual.

Done, thank you @AjeshRPai 🙇 

I'll set the milestone on this PR for... nevermind you did that 🥇 🚀

**Comment by AjeshRPai** [comment]
@ovitrif 

I have updated the Utils library version with this commit [78781ee](https://github.com/wordpress-mobile/WordPress-Android/pull/17477/commits/78781ee08fd235b9fa9ef9feb1e2e7c712b15e09). The Utils library version from the [CI build](https://buildkite.com/automattic/wordpress-utils-android/builds/145#01848ad1-eea8-4b0a-b9c5-e719c66605ee/165-264) is `trunk-591eefcd7599adf185cc33cc4098550e2f280454`. 

Enabling auto-merge.

**Pull request merged**
